src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs

```diff
@@ -267,15 +294,25 @@ public class AccessController : Controller
                 });
             }
 
+            var request = HttpContext.GetOpenIdConnectRequest();
```

Comment:
> as we need the request instance 

You mean, a parameter?

> it seems a bit ugly

Whatever you think is less ugly but would reuse the BadRequest code and message.